src/utils.py

```diff
@@ -22,6 +22,9 @@ def safe_write_to_file(content: str, path: str, mode: str = "w") -> None:
     with open(path, mode) as f:
         f.write(content)
 
+    # os.chmod(path, 0o770)
+    # shutil.chown(path, user="snap_daemon", group="root")
```

Comment:
Not sure I understand. Is this needed or not? If not I'd rather remove commented code

Reply:
If you find out let me know! Strange permission bug, am working to fix. Will move review to draft.